src/cloud-api-adaptor/test/provisioner/trustee_kbs.go

```diff
@@ -313,9 +427,33 @@ func decompressFileOnTargetNode(targetNodeIP, remoteFilePath, targetDir string)
 	return cmd.Run()
 }
 
-func NewBaseKbsInstallOverlay(installDir string) (InstallOverlay, error) {
-	log.Info("Creating kbs install overlay")
-	overlay, err := NewKustomizeOverlay(filepath.Join(installDir, "kbs/config/kubernetes/base/"))
+func NewHTTPSKbsInstallOverlay(installDir string) (InstallOverlay, error) {
+	log.Info("Creating https kbs install overlay")
+	if workerNodeIP != "" {
+		keyContent, certContent, err := generateCert(workerNodeIP)
+		fmt.Println("Certificate Content:")
+		fmt.Println(certContent)
+		fmt.Println("Key Content:")
+		fmt.Println(keyContent)
+
+		if err != nil {
+			fmt.Println("Error generating certificate and key:", err)
+		}
+	}
+
+	platform, err := getHardwarePlatform()
+	if err != nil {
+		return nil, err
+	}
+
+	var overlayFolder string
```

Comment:
Does the base overlay not provide the ability to use an HTTPS certificate? Do we have to use `CUSTOM_PCCS_URL` (what code is triggering this at the moment?), or nodeport?

Reply:
The code diff has gone now. I'm not sure what happened, but maybe the rebase resolved this?